test/algodclientv2_test.go

```diff
@@ -0,0 +1,379 @@
+		globalErrForExamination = nil
+	})
+}
+
+func weMakeAnyShutdownCall() error {
+	algodClient, err := algod.MakeClient(mockServer.URL, "")
+	if err != nil {
+		return err
+	}
+	globalErrForExamination = algodClient.Shutdown().Do(context.Background())
+	return nil
+}
+
+func weMakeAnyRegisterParticipationKeysCall() error {
+	algodClient, err := algod.MakeClient(mockServer.URL, "")
+	if err != nil {
+		return err
+	}
+	globalErrForExamination = algodClient.RegisterParticipationKeys("").Do(context.Background())
+	return nil
+}
+
+var stxResponse types.SignedTxn
+var pendingTransactionInformationResponse models.PendingTransactionInfoResponse
+
+func weMakeAnyPendingTransactionInformationCall() error {
+	algodClient, err := algod.MakeClient(mockServer.URL, "")
+	if err != nil {
+		return err
+	}
+	pendingTransactionInformationResponse, stxResponse, globalErrForExamination = algodClient.PendingTransactionInformation("").Do(context.Background())
+	return nil
+}
+
+func theParsedPendingTransactionInformationResponseShouldHaveSender(sender string) error {
+	if stxResponse.Txn.Sender.String() != sender {
+		return fmt.Errorf("expected txn to have sender %s but actual sender was %s", sender, stxResponse.Txn.Sender.String())
+	}
+	return nil
+}
+
+var stxsResponse []types.SignedTxn
+
+func weMakeAnyPendingTransactionsInformationCall() error {
+	algodClient, err := algod.MakeClient(mockServer.URL, "")
+	if err != nil {
+		return err
+	}
+	_, stxsResponse, globalErrForExamination = algodClient.PendingTransactions().Do(context.Background())
+	return nil
+}
+
+func theParsedPendingTransactionsInformationResponseShouldHaveSender(sender string) error {
+	if stxsResponse[0].Txn.Sender.String() != sender {
+		return fmt.Errorf("expected txn to have sender %s but actual sender was %s", sender, stxsResponse[0].Txn.Sender.String())
+	}
+	return nil
+}
+
+func theParsedPendingTransactionsInformationResponseShouldContainAnArrayOfLenAndElementNumberShouldHaveSender(length, idx int, sender string) error {
+	if len(stxsResponse) != length {
+		return fmt.Errorf("expected response length %d but received length %d", length, len(stxsResponse))
+	}
+	if stxsResponse[idx].Txn.Sender.String() != sender {
+		return fmt.Errorf("expected txn %d to have sender %s but real sender was %s", idx, sender, stxsResponse[idx].Txn.Sender.String())
+	}
+	return nil
+}
+
+var txidResponse string
+
+func weMakeAnySendRawTransactionCall() error {
+	algodClient, err := algod.MakeClient(mockServer.URL, "")
+	if err != nil {
+		return err
+	}
+	txidResponse, globalErrForExamination = algodClient.SendRawTransaction(nil).Do(context.Background())
+	return nil
+}
+
+func theParsedSendRawTransactionResponseShouldHaveTxid(txid string) error {
+	if txidResponse != txid {
+		return fmt.Errorf("expected txn to have txid %s but actual txid was %s", txidResponse, txid)
+	}
+	return nil
+}
+
+func weMakeAnyPendingTransactionsByAddressCall() error {
+	algodClient, err := algod.MakeClient(mockServer.URL, "")
```

Comment:
I'd probably pop this in an init function instead of repeating it in every step definition